### DISCUSSION TITLE
The array  tracked will become very slow when access

### DISCUSSION THREAD

**Discussion by yoodu** [pointfreeco/swift-perception#99] [General]
### Description

The code example is as follows. When using `@Observable`, it runs quickly

```
@Perceptible
final class AssetRow {
  var data: [Int] = []

  init(_ data: [Int]) {
    self.data = data
  }
}

print(Date(), 1)
let arr = (1 ... 3000).map { _ in AssetRow([]) }
print(Date(), 2)
let _ = arr.map { $0.data.count } // very slow
print(Date(), 3)
```
Result：
2024-09-15 14:04:16 +0000 1
2024-09-15 14:04:16 +0000 2
2024-09-15 14:04:22 +0000 3

### Checklist

- [X] I have determined that this bug is not reproducible using Swift's observation tools. If the bug is reproducible using the `@Observable` macro or another tool from the `Observation` framework, please [file it directly with Apple](https://github.com/apple/swift/issues/new/choose).
- [X] If possible, I've reproduced the issue using the `main` branch of this package.
- [X] This issue hasn't been addressed in an [existing GitHub issue](https://github.com/pointfreeco/swift-perception/issues) or [discussion](https://github.com/pointfreeco/swift-perception/discussions).

**Comment by mbrandonw** [comment]
Hi @yoodu, this is only true in debug builds because of the additional perception checks we perform at runtime. We try to detect when you access a field of a `@Perceptible` model while inside a view's `body` but _not_ inside `WithPerceptionTracking`. Those checks do take some [work](https://github.com/pointfreeco/swift-perception/blob/bc67aa8e461351c97282c2419153757a446ae1c9/Sources/Perception/PerceptionRegistrar.swift#L200-L223), but they are only run in debug. If you run your code in release you will find that there is no difference between `@Perceptible` and `@Observable`. And if you have ideas of how to speed up those checks we would happily take a PR.

Since this isn't an issue with the library I am going to convert it to a discussion. Please feel free to continue the conversation over there!
